src/renderer/features/assets/AssetsPortfolio/model/portfolio-model.ts

```diff
@@ -0,0 +1,174 @@
+    const tokens = tokensService.getTokensData();
+
+    return tokens.reduce((acc, token) => {
+      const filteredChains = token.chains.filter((chain) => {
+        return activeWallet?.accounts.some((account) => {
+          return (
+            accountUtils.isNonBaseVaultAccount(account, activeWallet) &&
+            accountUtils.isChainAndCryptoMatch(account, chains[chain.chainId])
+          );
+        });
+      });
+
+      if (filteredChains.length > 0) {
+        acc.push({ ...token, chains: filteredChains });
+      }
+
+      return acc;
+    }, [] as TokenAsset[]);
+  },
+);
+
+const populateTokensBalanceFx = createEffect(
+  ({
+    activeTokens,
+    balances,
+    accounts,
+  }: {
+    activeTokens: TokenAsset[];
+    balances: any;
+    accounts: Account[];
+  }): TokenAsset[] => {
+    return activeTokens.map((token) => {
+      let totalBalance = {} as TokenBalance;
+      const chainWithBalance = token.chains.map((chain) => {
+        const selectedAccountIds = accounts.reduce<AccountId[]>((acc, account) => {
+          if (accountUtils.isChainIdMatch(account, chain.chainId)) {
+            acc.push(account.accountId);
+          }
+
+          return acc;
+        }, []);
+
+        const accountsBalance = balanceUtils.getAssetBalances(
+          balances,
+          selectedAccountIds,
+          chain.chainId,
+          chain.assetId.toString(),
+        );
+
+        const assetBalance = accountsBalance.reduce<TokenBalance>((acc, balance) => {
+          return sumTokenBalances(balance, acc);
+        }, {} as Balance);
+
+        totalBalance = sumTokenBalances(assetBalance, totalBalance);
+
+        return { ...chain, balance: assetBalance };
+      });
+
+      return { ...token, chains: chainWithBalance, totalBalance };
+    });
+  },
+);
+
+sample({
+  clock: setActiveView,
+  target: $activeView,
+});
+
+sample({
+  clock: setAccounts,
+  target: $accounts,
+});
+
+sample({
+  clock: [walletModel.$activeWallet, $activeView, once(networkModel.events.networkStarted)],
+  source: {
+    activeView: $activeView,
+    activeWallet: walletModel.$activeWallet,
+    chains: networkModel.$chains,
+  },
+  filter: ({ activeView, activeWallet }) => {
+    return Boolean(activeView === AssetsListView.TOKEN_CENTRIC && activeWallet);
+  },
+  target: updateTokensFx,
+});
+
+sample({
+  clock: updateTokensFx.doneData,
+  target: $tokens,
+});
+
+sample({
+  clock: [networkModel.$connections, $tokens],
+  source: {
+    activeView: $activeView,
+    activeWallet: walletModel.$activeWallet,
+    connections: networkModel.$connections,
+    chains: networkModel.$chains,
+    tokens: $tokens,
+  },
+  filter: ({ connections, activeWallet, activeView }) => {
+    return Boolean(activeView === AssetsListView.TOKEN_CENTRIC && Object.keys(connections).length && activeWallet);
+  },
+  fn: ({ connections, chains, tokens, activeWallet }): TokenAsset[] => {
+    const isMultisig = walletUtils.isMultisig(activeWallet);
```

Comment:
I'm not sure in understand why we filter out multisigs here, and below with the `isMultisigSupported`? It seems weird to me that we filter out such chains, there are many.

Reply:
I took that part from the current chain filtration on the main page, I believe there was a point to add there in the first place 
Can't tell why it was added tbh

Reply:
maybe @tuul-wq  or @Asmadek know?

Reply:
@Tbaut It's made in order to process the following case:
1. The user selects multisig wallet
2. The user opens Portfolio page
3. Nova Spektr shouldn't show chains where multisig pallet doesn't exist because it's redundant.

Reply:
If wallet is not multisig we accept it, otherwise (if wallet is multisig) we check that multisig pallet exists. We don't need to check `isMultisig && hasPallet` directly here.